multiapps-controller-persistence/src/test/java/org/cloudfoundry/multiapps/controller/persistence/services/AccessTokenServiceTest.java

```diff
@@ -0,0 +1,180 @@
+        verifyAccessTokensAreEqual(ACCESS_TOKEN_2, accessTokens.get(1));
+        verifyAccessTokensAreEqual(ACCESS_TOKEN_3, accessTokens.get(2));
+    }
+
+    @Test
+    void testDeleteByOlderThan() {
+        accessTokenService.add(ACCESS_TOKEN_1);
+        accessTokenService.add(ACCESS_TOKEN_2);
+        assertEquals(2, accessTokenService.createQuery()
+                                          .olderThan(DATE_2)
+                                          .delete());
+    }
+
+    @Test
+    void testDeleteByOlderThanNothingDeleted() {
+        accessTokenService.add(ACCESS_TOKEN_1);
+        accessTokenService.add(ACCESS_TOKEN_2);
+        assertEquals(2, accessTokenService.createQuery()
+                                          .olderThan(DATE_2)
+                                          .delete());
+    }
+
+    @Test
+    void testUpdateToken() {
+        accessTokenService.add(ACCESS_TOKEN_1);
+        AccessToken accessToken = createAccessToken(1, getFakeTokenValue(), "nvvalchev@gmail.com", DATE_2);
+        AccessToken updateAccessToken = accessTokenService.update(ACCESS_TOKEN_1, accessToken);
+        verifyAccessTokensAreEqual(accessToken, updateAccessToken);
+        List<AccessToken> accessTokens = accessTokenService.createQuery()
+                                                           .list();
+        assertEquals(1, accessTokens.size());
+        verifyAccessTokensAreEqual(accessToken, accessTokens.get(0));
+    }
+
+    @Test
+    void testThrowExceptionOnConflictingEntity() {
+        accessTokenService.add(ACCESS_TOKEN_1);
+        Exception exception = assertThrows(ConflictException.class, () -> accessTokenService.add(ACCESS_TOKEN_1));
+        assertEquals("Access token with ID \"1\" already exist", exception.getMessage());
+    }
+
+    @Test
+    void testThrowExceptionOnEntityNotFound() {
+        Exception exception = assertThrows(NotFoundException.class, () -> accessTokenService.update(ACCESS_TOKEN_1, ACCESS_TOKEN_2));
+        assertEquals("Access token with ID \"1\" does not exist", exception.getMessage());
+    }
+
+    private AccessTokenService createAccessTokenService() {
+        EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("TestDefault");
+        AccessTokenService.AccessTokenMapper accessTokenMapper = new AccessTokenService.AccessTokenMapper();
+        return new AccessTokenService(entityManagerFactory, accessTokenMapper);
+    }
+
+    private static AccessToken createAccessToken(long id, byte[] value, String username, LocalDateTime expiresAt) {
+        return ImmutableAccessToken.builder()
+                                   .id(id)
+                                   .value(value)
+                                   .username(username)
+                                   .expiresAt(expiresAt)
+                                   .build();
+    }
+
+    private static byte[] getFakeTokenValue() {
```

Comment:
Why not used the value from JwtTokenParserTest.testParseJwtTokenWithNoExpirationTime(), without the real user mail there. Also you can directly use the token value here, instead of new file